### ISSUE TITLE
Message list scrolls to top when switching folders

### ISSUE THREAD

**Issue by oreineke** [RealRaven2000/QuickFolders#417]
With Quickfolders 6.2 on FF 115.3.2, when switching folders by clicking on a quick folders tab (same or different email account), in the selected folder the list of email messages often starts to scroll all the way to the top, unless you grab the scrollbar to stop it.

**Comment by RealRaven2000** [comment]
It _should_ scroll to the folder to make it visible (not to the top). Sometimes it doesn't "quite" get there. Anyway will investigate.

**Comment by oreineke** [comment]
It is not the folder pane that is scrolling, it is the email messages pane (i.e. the list with individual emails). It scrolls up through all my emails until it reaches the top of the email message list.

Maybe when switching folders via Quickfolders, the message pane tries to scroll to the message that was selected in the previous message pane - just a wild guess.

This wild scrolling does not happen if I select a folder trough the TB folder pane.

**Comment by oreineke** [comment]
I also noted that the undesired scrolling in the message pane does _not_ happen if you jump between quick folders using the "shift-j" shortcut.
It only happens when switching folders by clicking on the quick folders tabs.

**Comment by RealRaven2000** [comment]
On my system, it returns to the message that I had selected last time I visited the folder, so I cannot reproduce the behavior at them moment.

**Comment by oreineke** [comment]
I found it is important to have a message selected before switching folders to cause the scrolling. Also, the folder I am switching to seems to must have a lot of messages in it to show the symptom (I have ~2.500 in one). Maybe it is an overflow issue in some kind of message id?

**Comment by oreineke** [comment]
I found that the following code from quickfolders.js, Method "MySelectFolder" seems to be causing this issue:
```
    let doc = QuickFolders.Util.document3pane;
    if (doc) {
      doc.commandDispatcher.advanceFocus();
      doc.commandDispatcher.rewindFocus();
    }
```
If I disable one or both of the `...Focus()` calls, the crazy scrolling never seems to happen.

**Comment by RealRaven2000** [comment]
> I found that the following code from quickfolders.js, Method "MySelectFolder" seems to be causing this issue:
> 
> ```
>     let doc = QuickFolders.Util.document3pane;
>     if (doc) {
>       doc.commandDispatcher.advanceFocus();
>       doc.commandDispatcher.rewindFocus();
>     }
> ```
> 
> If I disable one or both of the `...Focus()` calls, the crazy scrolling never seems to happen.

thanks for chasing this up. it's been a while since this code has been written, I hardly remember, but i guess it is related to scrolling the message into view. it may well be obsolete in Thunderbird 115. Note that I have folders with > 15,000 messages but I usually sort by Date descending. Maybe we can drop these lines altogether.

You can disable the whole piece of code by setting 
`extensions.quickfolders.autoFocusPreview = false`
or through deactivating this setting:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/1dd74673-f01c-43f3-94b7-483a42c3c7f8)

come to think of it, it's switched off here. :)

**Comment by RealRaven2000** [comment]
Related (very old) bugs from the now decommissioned bugzilla which was hosted at mozdev.org:

https://quickfolders.org/bugzilla/bugs/show_bug.cgi@id=21808.html

that's all I could find, but obviously it's a very very old feature, as the bug is dated 2009 ...

**Comment by RealRaven2000** [comment]
OK, I am just playing with the  feature a little while I am in the same 3pane tab (just switching between different folders and putting the mouse into the folder pane manually. It seems like it probably affects the focus into the thread list, but doesn't really affect scrolling or selecting messages. So if it is just about focusing the message pane (I guess so we can scroll it) then I may find a different way of doing that. The question is whether a particular message needs to be focused in order to enable "space-bar" navigation?

**Comment by RealRaven2000** [comment]
So I have looked at the oldest history, this is pretty much an option from one of the first versions of QuickFolders (1.0.1) which was released on 15/05/2009:

https://quickfolders.org/version-old.html#1.0

Either it's time to retire the feature or reimplement it?

**Comment by RealRaven2000** [comment]
Can you try the version below? I mainly removed the calls to `commandDispatcher.advanceFocus()` and `commandDispatcher.rewindFocus() `. I also added some debug log which you can enable via QF options / debug mode / right click debug mode:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/9991e12c-07df-4984-a4c5-e05b890a9a6f)

by setting `extensions.quickfolders.debug.folders.select = true` there will be a little more detail in the Javascript console, as long as QuickFolders debug mode is activated.

-----

[QuickFolders-mx-6.3pre103.zip](https://github.com/RealRaven2000/QuickFolders/files/12927580/QuickFolders-mx-6.3pre103.zip)

To test the new version above, download the zip file, drag the file into Thunderbird Add-ons Manager, do not extract contents or if won't install.

**Comment by oreineke** [comment]
Thanks, the "QuickFolders-mx-6.3pre103.zip" test version does not show the unintended scrolling!
The message pane is not selected after switching folders, but that also did not work with the normal ESR115 Quickfolders release (with "Focus on message list when changing tab" option switched on). It seems to take 6 TAB presses to focus the message pane after switching folders, but also seems to depends on what other GUI elements are visible.

**Comment by RealRaven2000** [comment]
I have tried to improve the focusing of the message pane with the following version - I noticed that after focusing (which happens 200ms after clicking the QuickFolders Tab) the current mail is briefly selected but shortly after the focus goes elsewhere (not even know where it goes and how). But I added a longer delay (1000ms) and now it appears to work (for me). I highly recommend restarting Thunderbird after installing this version. Also enable "Focus on message list  when changing tabs"

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/7bb88e4d-a419-4ff0-bf1e-b17f5dc3e0b3)

this will trigger the improved behavior..

[QuickFolders-mx-6.3pre111.zip](https://github.com/RealRaven2000/QuickFolders/files/13030776/QuickFolders-mx-6.3pre111.zip)

----

To test the new version above, download the zip file, drag the file into Thunderbird Add-ons Manager, do not extract contents or if won't install.
